cmd/virt-launcher/safe-tail/safe-tail.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+logfile=$1
+socketfile=${logfile%-log}
+
+# Wait for the socket to appear
+i=0
+while [[ ! -S ${socketfile} && ${i} -le 30 ]]
+do
+  sleep 1
+  i=$(( ${i} + 1 ))
+done
+if [[ ! -S ${socketfile} ]]
+then
+  exit 2
+fi
+
+# Start the tail process in background and save its PID, force line buffer
+tail --quiet -F ${logfile} 2>/dev/null | grep --line-buffered "." 2>/dev/null &
+tailpid=$!
+
+# Trap relevant signals and forward them to the tail process
+trap 'kill -SIGTERM ${tailpid} 2>/dev/null; wait ${tailpid} 2>/dev/null' SIGINT SIGQUIT SIGPIPE SIGTERM
+
+# Quickly terminate the tail process when the socket file gets closed
+while [ -S ${socketfile} ]
```

Comment:
Does the socket file disappear when closed?

Reply:
I think we can safely assume that qemu is able to correctly handle a unix socket.
It's not an exhaustive test, but I tried to force kill the qemu process in a virt-launcher compute container with `kill -9` and the socket file instantaneously disappeared.

Reply:
ACK

Reply:
I'm still encountering some timeouts in the sig-compute-migrations lane so I also added a bit of explict cleanup logic in `virt-launcher-monitor`.